CHANGELOG.rst

```diff
@@ -4,6 +4,21 @@ Changelog
 Versions are year-based with a strict backward-compatibility policy.
 The third digit is only for regressions.
 
+24.1.0 (2024-01-25)
+-------------------
+Backward-incompatible changes:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- modified SSL.py to set SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER in calling OpenSSL. See cherrypy/cheroot#245 for discussion.
```

Comment:
We can reword this to emphasize the behavior change ("pyOpenSSL now sets `SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER` by default, matching CPython's behavior") over the implementation work that has been done. The specific file that has been modified is likely not as interesting for readers. :)

Also, it's probably useful to link (see below for syntax) to this PR here, which then has further links to the previous PR and the cherrypy discussion.

Reply:
Thanks! I made those changes you suggested. Regarding code coverage I noticed in SSL.py, one area coverage is missing is here:

```

172 | except AttributeError:
173 | # Hardcode constants for cryptography < 3.4, see
174 | # https://github.com/pyca/pyopenssl/pull/985#issuecomment-775186682
175 | SSL3_VERSION = 768
176 | TLS1_VERSION = 769
177 | TLS1_1_VERSION = 770
178 | TLS1_2_VERSION = 771
179 | TLS1_3_VERSION = 772

```

If we are requiring cryptography >= 42.0.0, can the exception block be replaced by `pass` or something?

Reply:
Thanks for spotting that! We can do away with the entire try/catch block(s) here and only keep the contents. On a second look that's probably what codecov is complaining about, I was mesmerized by the diff and didn't notice the "indirect changes" tab. 🫣

Reply:
Looks like that did the trick! Can you approve the merge?